### ISSUE TITLE
Image Calculator: improper output type conversion 

### ISSUE THREAD

**Issue by stelfrich** [knime-ip/knip#471]
When output type of the Image Calculator node is set to `UnsignedByteType`, results are not the same when the output type is set to e.g. `FloatType` and the conversion to `UnsignedByteType` is done subsequently via clipping.

See also http://forum.imagej.net/t/macro-help-for-simple-automated-image-arithmetic/5184/6

**Comment by imagejan** [comment]
As also noted on the forum thread, it is the responsibility of the caller to ensure that the result fits into the given type. (The Image Calculator cannot know what kind of calculation you'll do, and therefore cannot simply assume DoubleType or some such, as this wouldn't always be desirable. It's hard to follow the principle of least surprise here...)

It could be documented a bit better, though.

**Comment by stelfrich** [comment]
> It could be documented a bit better, though.

Indeed! Especially since we have the situation of an _Image Calculator_ menu item in ImageJ/Fiji with close ties to the ImageJ1 "world" and an _Image Calculator_ node using ImgLib2 under the hood. So I guess this should be clearly stated in the documentation of the _Image Calculator_ node.

**Comment by imagejan** [comment]
Well, we also have an ImageJ2 [`ImageCalculator`](http://javadoc.imagej.net/ImageJ/index.html?net/imagej/plugins/commands/calculator/ImageCalculator.html) that presumably acts the same as KNIP's ImageCalculator regarding the types. :-D

**Comment by stelfrich** [comment]
> Well, we also have an ImageJ2 ImageCalculator that presumably acts the same as KNIP's ImageCalculator regarding the types. :-D

As if the situation wasn't confusing enough already. :sob:

**Comment by stelfrich** [comment]
> It could be documented a bit better, though.

See #472.
